### PR TITLE
[NOLD-16] Added Twisted client and Websocket transport

### PR DESCRIPTION
**Pull request by cgrigis**

### CODE REVIEW

File: barrister/runtime.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
     :license: MIT, see LICENSE for more details.
 """
 
+from twisted.internet import defer
```
Comment (pierluca) [inline barrister/runtime.py:8]: adding this here, one needs to add this to the dependencies (setup.py / requirements.txt)

File: barrister/runtime.py [hunk 2]
```diff
@@ -487,7 +489,28 @@ def request(self, req):
           req
             List or dict representing a JSON-RPC formatted request
         """
-        return self.protocol.sendRPC(req)
+        self.deferred = defer.Deferred()
+
+        print("RPC --> {!r}".format(req))
```
Comment (pierluca) [inline barrister/runtime.py]: shouldn't we use the logging module instead?

File: barrister/runtime.py [hunk 3]
```diff
@@ -487,7 +489,28 @@ def request(self, req):
...
+    def response_received(self, payload):
+        """
+        Callback invoked when a response is received from the server.
+
+        :Parameters:
+          payload
+            The raw bytes received
+        """
+        message = json.loads(payload.decode('utf8'))
+        print("<-- RPC {!r}".format(message))
```
Comment (pierluca) [inline barrister/runtime.py]: logging ?

File: barrister/runtime.py [hunk 4]
```diff
@@ -530,9 +553,9 @@ def __init__(self, transport, validate_request=True, validate_response=True,
         self.validate_resp = validate_response
         self.id_gen = id_gen
 
-    def start(self):
+    def get_iface(self):
```
Comment (pierluca) [inline barrister/runtime.py]: it makes sense that internally this is called iface, but from a "user" perspective, shouldn't this really be the `get_api`?

**Review by pierluca — CHANGES_REQUESTED** [review]
LGTM but I wouldn't merge this into neurobat:master , if anything because we have a pending pull request with the Barrister guy from our master.

File: barrister/runtime.py (hunk 1 above)
Comment (cgrigis) [reply barrister/runtime.py:8]: - [x] Add Twisted as a dependency in `requirements.txt` and `setup.py`

File: barrister/runtime.py (hunk 2 above)
Comment (cgrigis) [reply barrister/runtime.py]: - [x] Use logging module instead of `print` statements

File: barrister/runtime.py (hunk 4 above)
Comment (cgrigis) [reply barrister/runtime.py]: - [x] Rename `get_iface()` to `get_api()`

**Pull request closed without merging**